crates/wallet/src/wallet/mod.rs

```diff
@@ -736,10 +736,10 @@ impl Wallet {
     }
 
     /// Get the next unused address for the given `keychain`, i.e. the address with the lowest
-    /// derivation index that hasn't been used.
+    /// derivation index that hasn't been used (hasn't been involved in any transaction).
```

Comment:
```suggestion
    /// derivation index that hasn't been used in a transaction.
```

Reply:
fixed in bcc5329afa8eae6493b54355063896be0f73ed9c